tensorboard/components/tensor_widget/types.ts

```diff
@@ -0,0 +1,211 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {BaseTensorHealthPill} from "./health-pill-types";
+
+/**
+ * The specs for deferred view into a tensor.
+ *
+ * A tensor is a typed, multi-dimensional array.
+ * This interface abstracts away the backing storage of the tensor value.
+ * It allows on-demand retrieval into any element or sub-array of the tensor.
+ */
+export interface TensorView {
+  /** Data type of the underlying tensor. */
+  dtype: string;
+
+  /** Shape of the underlying tensor. */
+  shape: number[];
+
+  /** Get a specific element. */
```

Comment:
I was a bit confused here; does `indices` refer to "coordinate" of a value of a tensor? At passing, I thought this method would return array of values based on indices. I think this is the parameter that needs some explaining (i.e., how it should not be bigger than the shape, for instance).

Reply:
Yes, `indices` refers to coordinates. Note that you need n indices to specify an element in an n-dimensional tensor. Added the info to the doc string.